lib/install/deps.js

```diff
@@ -804,6 +804,21 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
 
   if (tree.phantomChildren && tree.phantomChildren[pkg.name]) return null
 
+  // if any of the peer dependencies is a (dev) dependency of the current
+  // tree locations, we place the package here. This is a safe location
+  // where we don't violate the peer dependencies contract.
+  // It may not be the perfect location in some cases, but we don't know
+  // if (dev) dependencies are hoisted to another location yet, as they
+  // may not be loaded into the tree yet.
+  var peerDeps = pkg.peerDependencies
+  if (peerDeps) {
+    if (Object.keys(peerDeps).some(function (name) {
+      return deps[name] || devDeps[name]
```

Comment:
This should have a check where we only check `devDeps` if `tree.isTop` (see the `var devDeps ...` section just above this one). We think this is otherwise fine and actually a pretty nice patch!